### PR TITLE
[AE] Named Almost Everything in the Abe Class

### PR DESCRIPTION
**Pull request by THEONLYDarkShadow**
Named fields, flags, structures and functions in the `Abe` class. Also named an object type in the `BaseGameObject` class and a couple of fields in the `MusicController` class.

### CODE REVIEW

File: Source/AliveLibAE/Abe.hpp [hunk 1]
```diff
@@ -300,58 +300,58 @@ struct Abe_SaveState
...
+        eD4_Bit9_unused = 0x100,
+        eD4_Bit10_play_ledge_grab_sounds = 0x200,
+        eD4_Bit11_unused = 0x400,
+        eD4_Bit12_have_healing = 0x800,
         eD4_eBit13 = 0x1000,
-        eD4_eBit14 = 0x2000,
-        eD4_eBit15 = 0x4000,
-        eD4_eBit16 = 0x8000,
+        eD4_eBit14_is_mudanchee_vault_ender = 0x2000,
+        eD4_eBit15_is_mudanchee_vault_ender = 0x4000,
```
Comment (paulsapps) [inline Source/AliveLibAE/Abe.hpp:345]: 14 and 15 are the same? 🤔

File: Source/AliveLibAE/Abe.hpp [hunk 2]
```diff
@@ -300,58 +300,58 @@ struct Abe_SaveState
...
+        eD4_Bit4_unused = 0x8,
+        eD4_Bit5_prevent_chanting = 0x10,
+        eD4_Bit6_land_softly = 0x20,
+        eD4_Bit7_unused = 0x40,
+        eD4_Bit8_laugh_at_chant_end = 0x80,
+        eD4_Bit9_unused = 0x100,
+        eD4_Bit10_play_ledge_grab_sounds = 0x200,
+        eD4_Bit11_unused = 0x400,
+        eD4_Bit12_have_healing = 0x800,
         eD4_eBit13 = 0x1000,
```
Comment (paulsapps) [inline Source/AliveLibAE/Abe.hpp:343]: bit 13 not named?

Comment (THEONLYDarkShadow) [reply Source/AliveLibAE/Abe.hpp]: It connects to `Flags_114::e114_Bit10` in the `BaseAliveGameObject` class, so I'll tackle that one later.

File: Source/AliveLibAE/Abe.hpp (hunk 1 above)
Comment (THEONLYDarkShadow) [reply Source/AliveLibAE/Abe.hpp]: For some reason, I thought they were supposed to be the same. When I looked at the code again, I realised that one is supposed to be `eD4_eBit15_is_mudomo_vault_ender`. :sweat_smile: The fix has been pushed now.

**Pull request merged**